sintr_analysis_server_example/lib/query.dart

```diff
@@ -17,6 +17,10 @@ abstract class Mapper {
   /// A function used by the mapper to provide results to its client.
   AddResult addResult;
 
+  ///Flag used to signal jobs manager that mapper has finished its work
+  /// and [map] no longer needs to be called.
+  bool isMapComplete = false;
```

Comment:
Consider 'isMapStopped' rather than Complete? Makes it clearer that this can be premature with respect to the data.